qa/integration-tests/src/test/java/io/camunda/it/backup/BackupRestoreIT.java

```diff
@@ -0,0 +1,257 @@
+  //  private GenericContainer<?> storageContainer;
+  private String storagePath;
+  private BackupRestoreTestConfig config;
+
+  private void setupCamunda(final BackupRestoreTestConfig config) throws IOException {
+    //    startStorageContainer(config);
+    testStandaloneCamunda = new TestStandaloneCamunda();
+    this.config = config;
+    config.configure(testStandaloneCamunda, repositoryDir);
+    testStandaloneCamunda.start();
+    testStandaloneCamunda.awaitCompleteTopology();
+    createSearchClient();
+    createRepository(storagePath);
+  }
+
+  @AfterEach
+  public void afterEach() throws IOException {
+    testStandaloneCamunda.stop();
+    if (restClient != null) {
+      restClient.close();
+      esClient = null;
+    }
+    if (osRestClient != null) {
+      osRestClient.close();
+      opensearchClient = null;
+    }
+  }
+
+  public static Stream<BackupRestoreTestConfig> sources() {
+    return Stream.of(DatabaseType.ELASTICSEARCH)
+        .flatMap(
+            dbType ->
+                Stream.of(RepositoryType.values())
+                    .map(
+                        repositoryType ->
+                            new BackupRestoreTestConfig(dbType, repositoryType, "bucket"))
+                    .limit(1));
+  }
+
+  @ParameterizedTest
+  @MethodSource(value = {"sources"})
+  public void shouldBackupAndRestoreToPreviousState(final BackupRestoreTestConfig config)
+      throws IOException, InterruptedException {
+    // given
+    setupCamunda(config);
+    Thread.sleep(10000);
+    final var backupClient = testStandaloneCamunda.newBackupClient();
+    // a backup is requested
+    final var takeResponse = backupClient.takeBackup(1L);
+    assertThat(takeResponse)
+        .satisfies(Either::isLeft)
+        .extracting(r -> r.get().getScheduledSnapshots())
+        .asInstanceOf(InstanceOfAssertFactories.LIST)
+        .isNotEmpty();
+    final var snapshots = takeResponse.get().getScheduledSnapshots();
+    // the backup is completed
+    Awaitility.await("Backup completed")
+        .atMost(Duration.ofSeconds(600))
+        .until(
+            () -> {
+              final var backupResponse = backupClient.getBackup(1L);
+              return backupResponse.isRight()
+                  && backupResponse.get().getState() == BackupStateDto.COMPLETED
+                  && backupResponse.get().getDetails().stream()
+                      .allMatch(d -> d.getState().equals("SUCCESS"));
+            });
+
+    // then
+    // if we stop all apps and restart elasticsearch
+    testStandaloneCamunda.stop();
+    testStandaloneCamunda.startESContainer();
+    // perform a restore with a new client (old one is not valid anymore)
+    createSearchClient();
+    createRepository(storagePath);
+    deleteAllIndices();
+    restore(snapshots);
+  }
+
+  private void deleteAllIndices() throws IOException {
+    switch (config.databaseType) {
+      case ELASTICSEARCH -> {
+        esClient
```

Comment:
:wrench: I believe ES allows for something like `curl -XDELETE 'http://localhost:9200/*'` IIRC. Doesn't work on OpenSearch though :(